src/encoder.rs

```diff
@@ -1635,6 +1635,11 @@ pub fn encode_block_post_cdef<T: Pixel>(
           assert_eq!(mvs[0].col, mv_stack[0].this_mv.col);
         }
       }
+      if fi.is_filter_switchable
+        && ContextWriter::needs_interp_filter(bsize, luma_mode)
+      {
+        cw.write_interp_filter(w, tile_bo, bsize, 0, FilterMode::REGULAR);
```

Comment:
I think the reason why there was desync in your test is that the decoder does not think it needs to decode the filter for each inter mode block if a frame header does not have is_filter_switchable = true, i.e. instead decoder already has decoded default filter. In spec, https://aomediacodec.github.io/av1-spec/#inter-block-mode-info-syntax, there is if block by " if ( interpolation_filter == SWITCHABLE ) {", if true it decodes the filter.

Reply:
That's the fi.is_filter_switchable flag here I think.

Reply:
Right, I mean it should sync with 'is_filter_switchable' written in frame header. Otherwise, I think the decoder does not decode the info by cw.write_interp_filter(w, tile_bo, bsize, 0, FilterMode::REGULAR);  ?

Reply:
Right. I think it is synced, unless I'm missing something?

Reply:
I landed https://github.com/xiph/rav1e/commit/485d179206e3a777db95157a5b25a1a3b1991a8e#diff-e59bfbbede3bbaad0b099696e5ec5716L635 , and if fi.is_filter_switchable is true, encoder does not write defalut filter. Then, decoder knows it needs to decode filter from each inter mode block.